### PR TITLE
verbose:module/nomodule option implementation

### PR DESCRIPTION
**Pull request by theresa-m**
Fixes: https://github.com/eclipse/openj9/issues/9248
Docs issue: https://github.com/eclipse/openj9-docs/issues/593

Hotspot -verbose:module output format:
```
[0.034s][info][module,load] java.base location: jrt:/java.base
[0.035s][info][module,load] jdk.jlink location: jrt:/jdk.jlink
[0.035s][info][module,load] java.net.http location: jrt:/java.net.http
[0.035s][info][module,load] java.security.jgss location: jrt:/java.security.jgss
[0.035s][info][module,load] java.transaction.xa location: jrt:/java.transaction.xa
[0.036s][info][module,load] jdk.internal.jvmstat location: jrt:/jdk.internal.jvmstat
```

OpenJ9 output format (similar to OpenJ9 verbose:class):
```
module load: java.base from: jrt:/java.base
module load: jdk.dynalink from: jrt:/jdk.dynalink
module load: jdk.httpserver from: jrt:/jdk.httpserver
module load: jdk.unsupported from: jrt:/jdk.unsupported
module load: openj9.dtfj from: jrt:/openj9.dtfj
module load: jdk.jdwp.agent from: jrt:/jdk.jdwp.agent
module load: jdk.localedata from: jrt:/jdk.localedata
module load: jdk.zipfs from: jrt:/jdk.zipfs
```

Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

**Comment by theresa-m** [comment]
Can I have a review from a committer?

File: runtime/verbose/verbose.c
Comment (pshipton) [inline runtime/verbose/verbose.c:1375]: This memory needs to be freed when it doesn't fit into the buffer.

File: runtime/verbose/verbose.c
Comment (pshipton) [inline runtime/verbose/verbose.c:1141]: I don't understand why this case is needed. It seems the hooks are only registered when VERBOSE_SETTINGS_SET, and registering occurs after parsing all the options, so there is no case when the hooks need to be unregistered. parseVerboseArgumentList() is only called once, and registering the hooks isn't inside a loop.

**Comment by pshipton** [comment]
Pls test the case when a module is loaded from a jar file.

File: test/functional/cmdLineTests/verbosetest/playlist.xml
Comment (pshipton) [inline test/functional/cmdLineTests/verbosetest/playlist.xml:57]: We don't need to test all these modes. The code doesn't depend on JIT modes, GC policy, compressed or non-compressed. Testing NoOptions should be sufficient.

**Comment by theresa-m** [comment]
> Pls test the case when a module is loaded from a jar file.

Test output:

**$java -verbose:module --module-path helloworld.jar -m helloworld/HelloWorldApp**
module load: java.base from: jrt:/java.base
...
**module load: helloworld from: jrt:/helloworld**
module load: jdk.zipfs from: jrt:/jdk.zipfs
module load: java.desktop from: jrt:/java.desktop
module load: java.security.jgss from: jrt:/java.security.jgss
module load: jdk.jartool from: jrt:/jdk.jartool
module load: jdk.compiler from: jrt:/jdk.compiler
**Hello from HelloWorldApp**

File: runtime/verbose/verbose.c
Comment (pshipton) [inline runtime/verbose/verbose.c:646]: Is setting VERBOSE_SETTINGS_IGNORE or VERBOSE_SETTINGS_CLEAR still needed, I don't see these used.

Comment (pshipton) [reply runtime/verbose/verbose.c:646]: The reason it's needed for `nogc` and `nocvlass` is to support enabling/disabling verbose output at runtime.
https://docs.oracle.com/javase/8/docs/api/java/lang/management/ClassLoadingMXBean.html#setVerbose-boolean-
https://docs.oracle.com/en/java/javase/11/docs/specs/jvmti.html#SetVerboseFlag

Comment (theresa-m) [reply runtime/verbose/verbose.c]: Thanks I removed that line. The OPT_MODULE input is there so that `verbose:nomodule` is recognized.

Comment (pshipton) [reply runtime/verbose/verbose.c]: I just meant the line should be simplified, like follows. I don't think your change is correct, the test behavior shouldn't change.
`verboseOptions->module = VERBOSE_SETTINGS_IGNORE;`

Comment (theresa-m) [reply runtime/verbose/verbose.c]: hotspot doesn't seem to do anything to disable the module output when both `verbose:module` and `verbose:nomodule` are set. Thats likely a mistake though.

Comment (theresa-m) [reply runtime/verbose/verbose.c]: I'll add that line back in the simplified state.

Comment (pshipton) [reply runtime/verbose/verbose.c]: `-verbose:module,nomodule` does do the expected on Hotspot (no module output). I agree `-verbose:module -verbose:nomodule` showing module output seems like a Hotspot bug.

Comment (theresa-m) [reply runtime/verbose/verbose.c]: Updated.

**Comment by pshipton** [comment]
jenkins compile win jdk11

**Comment by pshipton** [comment]
jenkins compile win jdk8

**Comment by pshipton** [comment]
Ran the tests via grinder https://ci.eclipse.org/openj9/view/Test/job/Grinder/1354/

**Review by pshipton — APPROVED** [review]

**Pull request merged**
